data_model/src/transaction.rs

```diff
@@ -342,34 +350,53 @@ impl Transaction {
         self.payload.nonce = Some(nonce);
         self
     }
+}
 
-    /// Sign transaction with the provided key pair.
-    ///
-    /// # Errors
-    ///
-    /// Fails if signature creation fails
-    #[cfg(feature = "std")]
-    pub fn sign(
-        mut self,
-        key_pair: iroha_crypto::KeyPair,
-    ) -> Result<Transaction, iroha_crypto::Error> {
-        let signature = SignatureOf::new(key_pair, &self.payload)?;
-        self.signatures.insert(signature);
-
-        Ok(Self {
-            payload: self.payload,
-            signatures: self.signatures,
-        })
-    }
+/// Structure represents transaction in non-trusted form.
+///
+/// `Iroha` and its' clients use [`SignedTransaction`] to send transactions
+/// via network.  Direct usage in business logic is strongly
+/// prohibited. Before any interactions `accept`.
+#[version(n = 1, versioned = "VersionedSignedTransaction")]
+#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+pub struct SignedTransaction {
+    /// [`Transaction`] payload.
+    pub payload: Payload,
+    /// [`SignatureOf`] [`Payload`].
+    pub signatures: btree_set::BTreeSet<SignatureOf<Payload>>,
 }
 
-impl Txn for Transaction {
-    type HashOf = Self;
+macro_rules! impl_tx {
+    ($($ty:ty),+$(,)?) => {$(
+        impl Txn for $ty {
+            type HashOf = SignedTransaction;
 
-    #[inline]
-    fn payload(&self) -> &Payload {
-        &self.payload
-    }
+            #[inline]
+            fn payload(&self) -> &Payload {
+                &self.payload
+            }
+        }
+
+        #[cfg(feature = "std")]
+        impl SignableTransaction for $ty {
```

Comment:
What do you think about renaming to just `Sign`? It's more convinient in Rust.

We also know, that it is related to `Transaction` because it's inside `transaction` module